### PR TITLE
Fix #134

### LINKED ISSUE #134: Provide a way to know who is calling the service
Within the service operations ... To adequately support both the identity of the user and potential impersonation, it is vital that the fabric support picking up and relaying related headers for user identity, and ideally, any peer certificate information.

Depending on where OAuth goes with tokens, some of this pattern will obviously change in the future, but ultimately, services need to know WHO is making the call, or WHO the call is being made ON BEHALF of, so that they can make authorization decisions and route to dependent services accordingly.

Previously, this pattern was given ... which doesn't work since there while there is a peer on the context, there is no AuthInfo...
```
	p, ok := peer.FromContext(w)
	if ok {
		if p.AuthInfo != nil {
			tlsInfo := p.AuthInfo.(credentials.TLSInfo)
			subject = tlsutil.GetDistinguishedName(tlsInfo.State.VerifiedChains[0][0])
			// ... do stuff with subject (ie, setup clients with impersonation value)
		} else {
			// log or error if dependent on identifying user
		}
	} else {
		// log or error if dependent on identifying user
	}
```

Getting the peer from the context only exposes this potential AuthInfo, as well as the Addr of the peer.

If the USER_DN, EXTERNAL_SYS_DN, SSL_CLIENT_S_DN such headers are expected to be available on the context, then the appropriate helper functions need to be given for pulling those values.

### PR DESCRIPTION
**Pull request by alecholmez**
closes #134 

We now will pass HTTP without a `grpc-` prefix from the gateway. All headers will just be forwarded into the metadata context on the method. I also provided a helper function in the `rpcutil` package that will pull the `user_dn` string from the metadata.

### CODE REVIEW

File: rpcutil/pkg.go [hunk 1]
```diff
@@ -12,12 +12,49 @@
...
 func MatchHTTPHeaders(key string) (string, bool) {
 	key = textproto.CanonicalMIMEHeaderKey(key)
 	return key, true
 }
+
+// FetchDNFromContext will retrieve a user_dn from the incoming metadata stuffed in the gRPC method context
+func FetchDNFromContext(ctx context.Context) (string, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return "", errors.New("failed to retrieve rpc metadta from incoming context")
```
Comment (dougfort) [inline rpcutil/pkg.go]: I don't hate this enough to not approve. But IMO this an invalid use of the stdlib errors. I'd really like to see https://github.com/pkg/errors

Comment (alecholmez) [reply rpcutil/pkg.go]: Fair enough. I'll make an update

**Comment by lucasmoten** [comment]
ok. to clarify, this solution only handles the metadata headers, and requires something in front (nginx or other) to read client cert to populate those headers.  

We're still going to need to handle SSL connections somewhere if we want the whitelist support. 

i'm rebuilding my local project with these changes and updating tests to verify this can work with the headers

**Comment by alecholmez** [comment]
@lucasmoten That's correct. We should roll the old caller code into `tlsutil` if that's the case.

**Pull request merged**